src/Sarif.UnitTests/Visitors/InsertOptionalDataVisitorTests.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT        
+// license. See LICENSE file in the project root for full license information. 
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using FluentAssertions;
+using Microsoft.CodeAnalysis.Sarif.Readers;
+using Microsoft.CodeAnalysis.Sarif.UnitTests;
+using Newtonsoft.Json;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.Sarif.Visitors
+{
+    public class InsertOptionalDataVisitorTests : FileDiffingTests
+    {
+        // To rebaseline all test files set this value to true and rerun the testa
+        private static bool s_rebaseline = false;
+
+        private ITestOutputHelper _outputHelper;
+
+        private static string GetTestDirectory(string subdirectory = "")
+        {
+            return Path.GetFullPath(Path.Combine(@".\TestData", subdirectory));
+        }
+
+        private static string GetProductTestDataDirectory(string subdirectory = "")
+        {
+            return Path.GetFullPath(Path.Combine(@"..\..\..\..\..\\src\Sarif.UnitTests\TestData", subdirectory));
+        }
+
+        public InsertOptionalDataVisitorTests(ITestOutputHelper outputHelper)
```

Comment:
>ITestOutputHelper [](start = 46, length = 17)

In email you asked about adding diagnostic output to unit tests. That's what [`ITestOutputHelper`](https://xunit.github.io/docs/capturing-output) is for.

Reply:
indeed and i found it as you can see

---
In reply to: [205982298](https://github.com/Microsoft/sarif-sdk/pull/974#discussion_r205982298) [](ancestors = 205982298)